lib/reporting/proofing_rate_report.rb

```diff
@@ -86,43 +86,8 @@ def to_csv
 
     def reports
       @reports ||= begin
-        sub_reports = [0, *DATE_INTERVALS].each_cons(2).map do |slice_end, slice_start|
-          time_range = if slice_end.zero?
-                         Range.new(
-                           (end_date - slice_start.days).beginning_of_day,
-                           (end_date - slice_end.days).end_of_day,
-                         )
-                       else
-                         Range.new(
-                           (end_date - slice_start.days).beginning_of_day,
-                           (end_date - slice_end.days).end_of_day - 1.day,
-                         )
-                       end
-          Reporting::IdentityVerificationReport.new(
-            issuers: nil, # all issuers
-            time_range: time_range,
-            cloudwatch_client: cloudwatch_client,
-          )
-        end
-
-        reports = if parallel?
-                    threads = sub_reports.map do |report|
-                      Thread.new do
-                        report.tap(&:data)
-                      end.tap do |thread|
-                        thread.report_on_exception = false
-                      end
-                    end
-
-                    Reporting::UnknownProgressBar.wrap(show_bar: progress?) do
-                      threads.map(&:value)
-                    end
-                  else
-                    Reporting::UnknownProgressBar.wrap(show_bar: progress?) do
-                      sub_reports.each(&:data)
-                    end
-                  end
-
+        reports = parallel? ? parallel_reports : single_threaded_reports
```

Comment:
This kind of happened organically vs. an intentional refactor, as I ripped out the parallel logic and then added it back. But I like it -- the two paths shared almost no code.

Bikeshedding on `single_threaded_reports` as a name welcomed.